lib_xud/src/core/XUD_HAL.xc

```diff
@@ -148,12 +147,10 @@ void XUD_HAL_EnterMode_PeripheralFullSpeed()
     write_periph_word(USB_TILE_REF, XS1_SU_PER_UIFM_CHANEND_NUM, XS1_SU_PER_UIFM_FUNC_CONTROL_NUM, 
         (1<<XS1_SU_UIFM_FUNC_CONTROL_XCVRSELECT_SHIFT) | (1<<XS1_SU_UIFM_FUNC_CONTROL_TERMSELECT_SHIFT));
 #endif
-#endif
 }
 
 void XUD_HAL_EnterMode_PeripheralChirp()
 {
-#ifndef XUD_SIM_XSIM
 #ifdef __XS3A__
```

Comment:
In files that assume that no definition of `__XS3A__` assumes definition of `__XS2A__`, I would prefer to see near the top of the file:
```
#if !defined (__XS2A__) && !defined (__XS3A__)
#error Unknown architecture
#endif
```
That way the compiler complains when moving to new architectures rather than silently mixing together combinations of XS2 and XS3 functionality.